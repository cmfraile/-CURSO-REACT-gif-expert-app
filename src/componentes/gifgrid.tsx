import { v4 } from "uuid";

const placement = () => {
    let caso:any[] = [];
    for(let i = 1 ; i <= 5 ; i++){
        caso.push(
        <div    
                key={v4()}
                className="ggimg fondodefault"
                style={{backgroundImage:'url(https://c.tenor.com/txPXESQJ_0AAAAAC/the-bad-guys-the-bad-guys-movie.gif)'}}
        >
            <div className="titulo">
                Que ricos los papus
            </div>
        </div>
        )
    }
    return caso;
}

const getGifs = async(categoria:string) => {

    const querystring = ():string => {
        const request:any = {
            url:'https://api.giphy.com/v1/gifs/search?',
            api_key:'wOW1tP8JYE60hILwEOnpZ4bIQwDPu7mg',
            q:categoria,
            limit:'5'
        }
        const caso:string[] = [request.url];
        Object.keys(request).map(x => {
            if(x == 'url'){return};
            if(x == 'api_key'){caso.push(`${x}=${request[x]}`)}else{caso.push(`&${x}=${request[x]}`)}
            
        });
        return caso.join('');
    }
    const { data } = await(await fetch(querystring())).json();
    const gifs = data.map(({id,title,url}:any) => ({id,title,url}));
    //console.log(gifs);
}

export const GifGrid = ({categoria}:{categoria:string}) => {
    
    getGifs(categoria);

    return(
        <div className="gifgrid">
            <h3>{categoria}</h3>
            {placement()}
        </div>
    )
}