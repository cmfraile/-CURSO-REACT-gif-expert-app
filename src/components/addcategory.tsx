import { useState } from "react";
import propTypes from 'prop-types';

//export const AddCategory = ({sC}:{sC?:React.Dispatch<SetStateAction<string[]|[]>>}) => {
interface ACface {onNewCategory:any,cHK:string[]|[]};
export const AddCategory = ({onNewCategory,cHK}:ACface) => {

    const [inputValue,setInputValue] = useState<string>('')

    return(
        <form   aria-label='form'
                onSubmit={(e) => {
            //console.log('Hola mundo desde el submit');
            e.preventDefault();
            
            //VALIDADORES:
            if(inputValue.trim().length <= 2){return}
            if(cHK.map(x => x.toUpperCase()).includes(inputValue.trim().toUpperCase())){return};
            
            //sC((cat) => [inputValue,...cat]);
            onNewCategory(inputValue.trim())
            setInputValue('');
        }}>
            <input  type="text"
                    value={inputValue}
                    aria-label='inputcat'
                    placeholder="buscar gifs"
                    onChange={(e) => {setInputValue(e.target.value)}}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory:propTypes.func.isRequired
}
