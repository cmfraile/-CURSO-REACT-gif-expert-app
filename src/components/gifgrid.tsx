import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from 'prop-types';

export const GifGrid = ({categoria}:{categoria:string}) => {

    const { images, isLoading } = useFetchGifs(categoria);

    return(

        <div className="gifgrid">
            <h3 aria-label='title' >{categoria}</h3>
            <h2 style={{display:(isLoading) ? '' : 'none'}}>Cargando...</h2>
            {images.map(({id,title,url}) => (
            <div
                aria-label='imgcard'    
                key={id}
                className="ggimg fondodefault"
                style={{backgroundImage:`url(${url})`}}
            >
            <div className="titulo" style={{display:(title) ? '' : 'none' }} >{title}</div>
            </div>
            ))}
        </div>
    )
}

GifGrid.propTypes = { categoria : PropTypes.string.isRequired };