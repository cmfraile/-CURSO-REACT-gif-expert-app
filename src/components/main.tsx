import './main.sass';
import { useState } from 'react';
import { v4 } from 'uuid';
import { AddCategory } from './addcategory';
import { GifGrid } from './gifgrid';

export const main = () => {

    const [ categoriasHook , setCategorias ] = useState<string[]|[]>([]);

    return(
        <>
            <h1>Gif Expert APP</h1>
            
            <AddCategory
                // sC={setCategorias}
                onNewCategory={(e:string) => {setCategorias([e,...categoriasHook])}}
                cHK={categoriasHook}
            />

            <ol>{categoriasHook.map(valor => {return <GifGrid key={v4()} categoria={valor}/>})}</ol>

            {/*Lista de gifs*/}
                {/*Gif cards*/}
        </>
    )
}