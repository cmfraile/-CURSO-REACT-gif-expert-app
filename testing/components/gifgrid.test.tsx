import { render , screen } from '@testing-library/react';
import { GifGrid } from '../../src/components/gifgrid';
import React from 'react';

const main = () => {

    const valorcat:string = 'JoJo';

    test('Screenshot',() => {
        expect(render(<GifGrid categoria={valorcat}/>).container).toMatchSnapshot();
    });

    test('debe de aparecer el tÃ­tutlo de categoria correctamente insertado',() => {
        expect(render(<GifGrid categoria={valorcat}/>));
        expect(screen.getByLabelText('title').innerHTML).toContain(valorcat);
    });

};

describe('Pruebas en el <GifGrid>',main)