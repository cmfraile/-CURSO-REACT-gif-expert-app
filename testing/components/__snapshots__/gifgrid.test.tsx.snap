// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid> Screenshot 1`] = `
<div>
  <div
    class="gifgrid"
  >
    <h3
      aria-label="title"
    >
      JoJo
    </h3>
    <h2>
      Cargando...
    </h2>
  </div>
</div>
`;
